### ISSUE TITLE
Cambridge test suite is failing on my machine

### ISSUE THREAD

**Issue by KevinRansom** [dotnet/fsharp#10148]
It seems to pass the ci, but I don't know what the difference is.  And my machine is set up, how I think it should be set up.  So I'm guessing I'm not the only one, who sees these failures.

````
  X controlChamenos-FSC_BASIC [4s 683ms]
  Error Message:
   Error running command 'dotnet.exe' with args 'run -f netcoreapp3.1' in directory 'C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz'.
---- stdout below ---
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018: The "GenerateDepsFile" task failed unexpectedly. [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified. [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018: File name: 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018:    at Microsoft.NET.Build.Tasks.FrameworkReferenceResolver.GetDefaultReferenceAssembliesPath() [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GenerateDepsFile.WriteDepsFile(String depsFilePath) [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GenerateDepsFile.ExecuteCore() [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\Users\codec\AppData\Local\Temp\FSharp.Cambridge\l2f0e4nz.bdz\nzcuz55i.vvm.fsproj]

---- stderr below ---

The build failed. Fix the build errors and run again.
 . ERRORLEVEL 1
  Stack Trace:
     at TestFramework.checkResult(CmdResult result) in C:\kevinransom\fsharp\tests\FSharp.Test.Utilities\TestFramework.fs:line 313
   at SingleTest.executeFsc@270.Invoke(String testCompilerVersion, String targetFramework) in C:\kevinransom\fsharp\tests\fsharp\single-test.fs:line 278
   at SingleTest.executeSingleTestBuildAndRun@220.Invoke(OutputType outputType, String compilerType, String targetFramework, Boolean optimize, Boolean buildOnly) in C:\kevinransom\fsharp\tests\fsharp\single-test.fs:line 280
   at SingleTest.singleTestBuildAndRunCore(TestConfig cfg, String copyFiles, Permutation p, String languageVersion) in C:\kevinransom\fsharp\tests\fsharp\single-test.fs:line 303
   at SingleTest.singleTestBuildAndRunAux(TestConfig cfg, Permutation p) in C:\kevinransom\fsharp\tests\fsharp\single-test.fs:line 364
   at FSharp.Tests.Core.CoreTests.controlChamenos-FSC_BASIC() in C:\kevinransom\fsharp\tests\fsharp\tests.fs:line 468


````

**Comment by abelbraaksma** [comment]
Currently, all tests work on my machine, provided in some cases I add `-ci -nobl` to the `build` command and **only** in `release` build, i.e.:

```f#
build -testCambridge -c Release -ci -nobl
```

Is that what fails for you too? 

My local copy is a little behind. I could update date-by-date and rerun until it fails so we can pinpoint the exact commit that caused this.

**Comment by baronfel** [comment]
I have a similar symptom though not with the cambridge suite, instead on the `FSharp.Compiler.ComponentTests` suite running for the `netcoreapp3.1` TFM. Like @KevinRansom the failure is due to not being able to find the `Microsoft.DotNet.PlatformAbstractions, Version 2.1.0.0` reference when discovering the SDK locations, but I've verified that the dll does live in my nuget package cache.  I've attached the binlog from the build that the tests trigger to help investigate.


[binlog.zip](https://github.com/dotnet/fsharp/files/5471409/binlog.zip)
